pkg/cmd/release/create/gitResources_test.go

```diff
@@ -0,0 +1,414 @@
+﻿package create_test
+
+import (
+	"bytes"
+	"errors"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/OctopusDeploy/cli/pkg/cmd/release/create"
+	"github.com/OctopusDeploy/cli/test/testutil"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestReleaseCreate_ParseGitResourceOverrideString(t *testing.T) {
+	tests := []struct {
+		input           string
+		expect          *create.GitResourceGitRef
+		isErrorExpected bool
+	}{
+		//Valid inputs
+		{input: "Action:main", expect: &create.GitResourceGitRef{ActionName: "Action", GitRef: "main", GitResourceName: ""}},
+		{input: "Action:*", expect: &create.GitResourceGitRef{ActionName: "Action", GitRef: "*", GitResourceName: ""}},
+		{input: "Action=refs/heads/main", expect: &create.GitResourceGitRef{ActionName: "Action", GitRef: "refs/heads/main", GitResourceName: ""}},
+		{input: "Action=*", expect: &create.GitResourceGitRef{ActionName: "Action", GitRef: "*", GitResourceName: ""}},
+		{input: "Action:Name1:main", expect: &create.GitResourceGitRef{ActionName: "Action", GitRef: "main", GitResourceName: "Name1"}},
+		{input: "Action:Name1:*", expect: &create.GitResourceGitRef{ActionName: "Action", GitRef: "*", GitResourceName: "Name1"}},
+		{input: "Action=Name1=refs/heads/main", expect: &create.GitResourceGitRef{ActionName: "Action", GitRef: "refs/heads/main", GitResourceName: "Name1"}},
+		{input: "Action=Name1=*", expect: &create.GitResourceGitRef{ActionName: "Action", GitRef: "*", GitResourceName: "Name1"}},
+
+		//Invalid inputs
+		{input: "", isErrorExpected: true},
+		{input: "    ", isErrorExpected: true},
+		{input: "Action", isErrorExpected: true},
+		{input: ":refs/heads/main", isErrorExpected: true},
+		{input: "::refs/heads/main", isErrorExpected: true},
```

Comment:
What happens if you mix the delimiters? eg `Action=Name1:main`?

Reply:
I _think_ it actually still works. I'll add a test and check

Reply:
It does work, but it's not consistent with the server-side code (which only supports using one delimiter type at once).
Updated in adc6c26 to be consistent. Also learned what `rune` represents (`char`)

Reply:
IKR, it's so weird